### PR TITLE
Bugfix/756 disable retry nextcloud image loading

### PR DESCRIPTION
**Pull request by absurd-turtle**
Disabled the retry on images in the plant list.
I also disabled the error messages on images from the plant list because currently there is no way to check which plants have an icon already.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: frontend/src/features/map_planning/layers/plant/components/PlantListItem.tsx [hunk 1]
```diff
@@ -34,6 +34,8 @@ export function PlantListItem({ plant, onClick, isHighlighted = false }: PlantLi
           defaultImageUrl={defaultImageUrl}
           path={`Icons/${plant?.unique_name}.png`}
           shareToken="2arzyJZYj2oNnHX"
+          retry={false}
```
Comment (Bushuo) [inline frontend/src/features/map_planning/layers/plant/components/PlantListItem.tsx:37]: Did you try something like this? Maybe it makes sense to retry in the other cases.
Code does not work here, but such a function can be passed to the `useQuery` hook.
```ts
          retry={(failureCount, error) => {
              if(check if error is network 404 error) return false;
              return true;
          }}
```

File: frontend/src/features/nextcloud_integration/components/PublicNextcloudImage.tsx [hunk 2]
```diff
@@ -7,6 +7,10 @@ interface PublicNextcloudImageProps extends React.ComponentPropsWithoutRef<'img'
   shareToken: string;
   // placeholder that will be displayed if the requested image was not found
   defaultImageUrl?: string;
+  // Whether fetching the image should be retried on fail.
+  retry: boolean;
```
Comment (Bushuo) [inline frontend/src/features/nextcloud_integration/components/PublicNextcloudImage.tsx:11]: Maybe pass the retry type from react query here for flexibility

File: frontend/src/features/nextcloud_integration/hooks/usePublicImage.tsx [hunk 3]
```diff
@@ -14,6 +14,8 @@ type UsePublicImageOptions = {
   fallbackImageSource?: string;
   /** Whether an error modal should be displayed if the image can't be loaded. */
   showErrorMessage?: boolean;
+  // Whether fetching the image should be retried on fail.
+  retry: boolean;
```
Comment (Bushuo) [inline frontend/src/features/nextcloud_integration/hooks/usePublicImage.tsx:18]: Here as well.

**Review by Bushuo — APPROVED** [review]
Good work, but please try my suggestion.

**Comment by markus2330** [comment]
jenkins build please

**Review by markus2330 — CHANGES_REQUESTED** [review]
Thank you for the PR! Frontend build unfortunately fails. Please also check out @Bushuo's improvement suggestions.

**Comment by absurd-turtle** [comment]
> Thank you for the PR! Frontend build unfortunately fails. Please also check out @Bushuo's improvement suggestions.

I saw there are still some issues. Will fix it today.

**Comment by markus2330** [comment]
The user study is already 17:00, so the earlier this can be fixed, the better :rocket:

**Comment by markus2330** [comment]
Great work, thank you!

Are you working on debouncing or should we release as is? (If not further bugs are found.)

**Comment by absurd-turtle** [comment]
> Great work, thank you!
> 
> Are you working on debouncing or should we release as is? (If not further bugs are found.)

I might look into it today but probably won't finish. So not happening in this release.

**Pull request merged**